Client/Application/Migration.swift

```diff
@@ -26,8 +26,25 @@ class Migration {
     }
     
     static func moveDatabaseToApplicationDirectory() {
-        //Moves Coredata sqlite file from Documents dir to application support dir.
-        DataController.shared.migrateToNewPathIfNeeded()
+        if Preferences.Database.DocumentToSupportDirectoryMigration.completed.value {
```

Comment:
Can we merge the two if's and the code below. and keep a common comment for it.

Reply:
Going to leave separate to leave comment clarity. Otherwise comment will be something like:
// this 
// or
// something else

Which is okay, but given how sensitive this is, I think this is probably fine. Appreciate the feedback though.